packages/replay/src/auth.ts

```diff
@@ -263,3 +282,115 @@ export async function maybeAuthenticateUser(options: Options = {}) {
     return false;
   }
 }
+
+async function getAuthInfo(key: string): Promise<string> {
+  const resp = await query(
+    "AuthInfo",
+    `
+        query AuthInfo {
+          viewer {
+            user {
+              id
+            }
+          }
+          auth {
+            workspaces {
+              edges {
+                node {
+                  id
+                }
+              }
+            }
+          }
+        }
+      `,
+    undefined,
+    key
+  );
+
+  if (resp.errors) {
+    throw new GraphQLError("Failed to fetch auth info", resp.errors);
+  }
+
+  const response = resp.data as {
+    viewer: {
+      user: {
+        id: string | null;
+      } | null;
+    };
+    auth: {
+      workspaces: {
+        edges: {
+          node: {
+            id: string;
+          };
+        }[];
+      };
+    };
+  };
+
+  const { viewer, auth } = response;
+
+  if (viewer?.user?.id) {
+    viewer.user.id;
+  }
+
+  if (auth?.workspaces?.edges?.[0]?.node?.id) {
+    return auth.workspaces.edges[0].node.id;
+  }
+
+  throw new Error("Unrecognized type of an API key: Missing both user ID and workspace ID.");
+}
+
+function getAuthInfoCachePath(options: Options = {}) {
+  const directory = getDirectory(options);
+  return path.resolve(path.join(directory, "profile", "authInfo.json"));
+}
+
+function authInfoCacheKey(key: string) {
+  return createHash("sha256").update(key).digest("hex");
+}
```

Comment:
A code comment explaining this could be helpful. I **assume** that we want to hide the actual API key in the file on disk **if** the user didn't use `login` (in such a case the API key is written on disk anyway) but a different way of supplying the API key (like an env variable)